README.md

```diff
@@ -1,46 +1,123 @@
-7. Generate URLs with route helpers that require variables.
 
+## Paths vs Route Helpers
 
-## Notes
+So why would we want to use route helper methods as opposed to hard coding paths into the application? There are a number of reasons, below are a few of the key rationales:
 
-Given a posts#index and a posts#show, they should link between each other
+* Route helper methods help clean up the view and controller code and assist with readability. On a side note, you cannot use these helper methods in your model files.
 
-we could use `<a href="/posts/#{@post.id}">` but thats very low level.
+* It's more natural to be able to pass arguments into a method as opposed to using string interpolation. For example, using ```post_path(post, opt_in: true)``` is more readable than using ```"posts/<%= post.id %>?opt_in=true"```
 
-Let's start by having posts#index link to individual posts show pages.
+* Route helpers are more dynamic since they are methods and not simply strings
 
-link_to instead of a href but lets still use interpolation.
+* Route helpers translate directly into HTML friendly paths
 
-Then let's introduce the idea of named routes and instead of hardcoding URLs we'll let our URL Helpers generate them.
 
-what are the _path / _url helpers and when / where in teh stack can you use them (views/controllers yes, models no)
+## Implementing Route Helpers
 
-Why? Maintainability, relying on abstractions.
+To begin, we're going to start with an application that has the MVC setup for ```posts```, with ```index``` and ```show``` actions currently in place. The route call looks like this:
 
-rake routes
+```ruby
+# config/routes.rb
+resources :posts, only: [:index, :show]
+```
 
-how routes get names, the default
+We briefly discussed this ```resources``` method in the dynamic routing lesson, this will create routing methods for posts that we can utilized in our views and controllers. By running ```rake routes``` in the terminal it will give the following output:
 
-works great for posts_path , bad for /posts/:id
+```
+posts   GET  /posts(.:format)       posts#index
+post    GET  /posts/:id(.:format)   posts#show
+```
 
-introduce the :as option to explictly name a route
+These four columns tell us everything that we'll need in order to use the route helper methods. The breakdown is below:
 
-common error of trying to generate a route with a variable without providing an argument for that route variabel value.
+* **Column 1** - This column gives the prefix for the route helper methods. In the current application ```posts``` and ```post``` are the prefixes for the methods that you can use throughout your applications. The two most popular method types are ```_path``` and ```_url```. So if we want to render a relative link path to our posts' index page the method would be ```posts_path``` or ```posts_url```. The difference between ```_path``` and ```_url``` is that ```_path``` gives the relative path and ```_url``` renders the full URL.
```

Comment:
I would show some example code with what each thing returns so tell them to open up `rails console` and type

```
posts_path
```
 and all the other things to show the differences